### PR TITLE
Stashes

### PR DESCRIPTION
**Pull request by NanakoAC**
This PR adds Stashes, a story driven treasure hunting minigame. What are they?

It starts with a note. Crumpled notes are found hidden among maintenance loot. Each note tells a brief story of someone from eris' past, written from their perspective and vaguely talking about a time in their life when they had a need to hide something.  There are thieves and pirates stashing their stolen loot, mutineers stockpiling weapons, bootleggers gathering booze, and deposed captains trying to protect their last personal effects.

Most of the notes are written by Nikov, aka Maanfryd, our lovely lore advisor.

Within each note is a clue, which leads to a pseudo-randomly chosen location. There are three possible clues:

Stash landmark: This is the exact same system used to spawn excelsior starting gear. The clue is a text description of a place somewhere on the ship

Map: A photo of the location, that you have to identify by its surroundings. This uses a new Historical photo mode i added, which only captures turfs and anchored objects. It's assumed that unanchored items and mobs are recent additions and don't feature in the picture.

Coords: The least common clue, just x/y/z coords to a location

Stashes are always hidden under floor tiles at their location if possible, which makes them very hard to find if you don't know they're there. the chance of stumbling on a stash accidentally is very very low

Each stash contains a large quantity of uncommon or rare stuff designed around various themes. Some contain a ton of booze, some have gold, diamonds and credits, some are packed with guns and ammo. A couple of the rarer ones contain certain antag and command gear. They are a veritable jackpot for the adventurous assistant who seeks them out, but they are pretty rare and it can be hard to locate it

Various random values are tweaked to get an average of eight stashes spawning at roundstart. That's not much, scattered across a huge ship, so they're pretty uncommon

### CODE REVIEW

File: code/_helpers/global_lists.dm [hunk 1]
```diff
@@ -182,6 +192,26 @@ var/global/list/unworn_slots = list(slot_l_hand,slot_r_hand, slot_l_store, slot_
...
+		if (L.type == L.base_type)
+			//This is a base category. Add it to the categories list with a weighting
+			GLOB.stash_categories[L.base_type] = L.weight
+
+
+
+
+		else
+			//This is a specific stash datum, add it to the appropriate sublist
+			world << "Storing [L] [L.type] in all stash datums"
```
Comment (seekin4u) [inline code/_helpers/global_lists.dm]: world debug in master reeeeeeeeee

File: code/_helpers/global_lists.dm [hunk 2]
```diff
@@ -182,6 +192,26 @@ var/global/list/unworn_slots = list(slot_l_hand,slot_r_hand, slot_l_store, slot_
 		var/datum/job/J = new T
 		joblist[J.title] = J
 
+	//Stashes
+	paths = typesof(/datum/stash)-/datum/stash
```
Comment (seekin4u) [inline code/_helpers/global_lists.dm]: `paths = subtypesof(/datum/stash)`

File: code/modules/paperwork/paper.dm [hunk 3]
```diff
@@ -407,41 +407,43 @@
...
+					h_user.r_store = B
+					h_user.update_inv_pockets()
+				else if (h_user.head == src)
+					h_user.u_equip(src)
+					h_user.put_in_hands(B)
+				else if (!istype(src.loc, /turf))
+					src.loc = get_turf(h_user)
+					if(h_user.client)	h_user.client.screen -= src
+					h_user.put_in_hands(B)
+			user << "<span class='notice'>You clip the [P.name] to [(src.name == "paper") ? "the paper" : src.name].</span>"
```
Comment (seekin4u) [inline code/modules/paperwork/paper.dm]: don't we have SPAN_NOTICE() for things like that?

**Comment by NanakoAC** [comment]
Fixed requested changes

**Pull request merged**